### PR TITLE
search list component added

### PR DESCRIPTION
**Pull request by Satyam1203**

### CODE REVIEW

File: App.tsx [hunk 1]
```diff
@@ -15,6 +15,7 @@ const routes = {
   Alternate: "Alternate",
   Test: "Test",
   First: "First",
+  List: "List"
```
Comment (saurshaz) [inline App.tsx:18]: This is a component, do'n create a route for this.

File: src/pages/List.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
+import { useNavigation } from "@react-navigation/native";
```
Comment (saurshaz) [inline src/pages/List.js:1]: Move this to components dir

File: src/pages/List.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export const List = () => {
+  const { goBack } = useNavigation();
+  const [data, setData] = useState([]);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    setLoading(true);
+    const fetchData = async () => {
+      const res = await fetch('https://run.mocky.io/v3/56a7c1e0-434a-4773-b6b4-4cfc12fe1624');
```
Comment (saurshaz) [inline src/pages/List.js:16]: entire fetchData function shall be received as a prop

File: src/pages/List.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+  )
+
+  return (
+    <View style={styles.container}>
+        <Text style={styles.heading}>Search Filter Example</Text>
+        <SearchList 
+            data={data} 
+            searchFields={["name", "description", "category", "subCategory"]} 
+            visibleKeys={["name", "category", "subCategory"]}
+            titleStyle={null}
```
Comment (saurshaz) [inline src/pages/List.js:38]: get the  titleStyle and data style from props

File: src/pages/List.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+            dataStyle={{color: 'darkblue'}}
+        />
+        <Button
+            style={styles.link}
+            accessibilityRole="link"
+            onPress={() => {
+            goBack();
+            }}
+            title="Go Back"
```
Comment (saurshaz) [inline src/pages/List.js:47]: get the  style and title from props

File: src/pages/components/Header.tsx [hunk 6]
```diff
@@ -18,6 +18,9 @@ export default function Header() {
       <Link style={{ color: "green", fontSize: 20 }} to="/Test">
         Test
       </Link>
+      <Link style={{ color: "green", fontSize: 20 }} to="/List">
```
Comment (saurshaz) [inline src/pages/components/Header.tsx:21]: Don't add this in Header

File: src/pages/components/SearchList/index.js [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+  const filterData = data.filter(
+    createFilter(searchItem, searchFields)
+  )
+
+  const keys = visibleKeys || Object.keys(data[0] || []);
+
+  return (
+    <View>
+        <View style={{ flexDirection: "row", justifyContent: "space-around" }}>
```
Comment (saurshaz) [inline src/pages/components/SearchList/index.js:16]: try to extract all varying details like search placeholder, style of containers/texts  to be coming from props

**Review by saurshaz — CHANGES_REQUESTED** [review]
Minor refactoring requested @Satyam1203

File: App.tsx (hunk 1 above)
Comment (Satyam1203) [reply App.tsx:18]: removed it

File: src/pages/components/SearchList/index.js (hunk 7 above)
Comment (Satyam1203) [reply src/pages/components/SearchList/index.js:16]: added inputPlaceholder, flexWidth, numberOfLines, other styling stuffs to be coming from props

**Pull request merged**